src/main/scala/io/computenode/cyfra/dsl/GSeq.scala

```diff
@@ -119,12 +117,12 @@ object GSeq:
     val streamNextExpr = seq.source.next 
     val seqExprs = seq.elemOps.map(_.fn)
     
-    val limitExpr = ConstInt32(seq.limit.getOrElse(throw new IllegalArgumentException("Reduce on infinite stream is not supported")))
+    // Ensure a default limit to avoid infinite stream errors
+    val defaultLimit = 100 // Set a safe default limit
+    val limitExpr = ConstInt32(seq.limit.getOrElse {
+      println("⚠ Warning: No limit set, using default limit of " + defaultLimit)
```

Comment:
It's to be removed after addressing previous comment, but regardless, if we are to introduce warnings it's not a good idea to `println`. You usually want to use some logging API that can be configured/adjusted by library user, SLF4J is a good example.